### ISSUE TITLE
3.10.0 build failure on macOS: 'strdup' was not declared in this scope

### ISSUE THREAD

**Issue by ilovezfs** [ablab/spades#12]
Tested on macOS 10.10, 10.11, and 10.12: https://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/6988/

The error is
```
[ 83%] Building CXX object common/utils/CMakeFiles/utils.dir/autocompletion.cpp.o
cd /tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/build/common/utils && /usr/local/Homebrew/Library/Homebrew/shims/super/g++-6   -DNDEBUG -DUSE_GLIBCXX_PARALLEL=1 -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/build/common/utils -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common/utils -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/include -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/build/include -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src -I/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common -isystem /tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/../ext/include -isystem /usr/local/include  -fopenmp -std=c++11 -DNDEBUG   -Wno-deprecated -g0 -O2 -Wall -Wextra -Wconversion -Wno-sign-conversion -Wno-long-long -Wwrite-strings -o CMakeFiles/utils.dir/autocompletion.cpp.o -c /tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common/utils/autocompletion.cpp
/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common/utils/autocompletion.cpp: In function 'char* online_visualization::CommandGenerator(const char*, int)':
/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common/utils/autocompletion.cpp:22:72: error: 'strdup' was not declared in this scope
             char* answer = strdup(list_possible_matches.front().c_str());
                                                                        ^
/tmp/spades-20170202-70042-4p7f29/SPAdes-3.10.0/src/common/utils/autocompletion.cpp:30:45: error: 'strlen' was not declared in this scope
             if (!name.compare(0, strlen(text), text))
                                             ^
make[2]: *** [common/utils/CMakeFiles/utils.dir/autocompletion.cpp.o] Error 1
make[1]: *** [common/utils/CMakeFiles/utils.dir/all] Error 2
make: *** [all] Error 2
```

As a workaround, I can toss in `#include <string.h>` just after the `#include <string>`:
```
    inreplace "src/common/utils/autocompletion.cpp", /(#include <string>)/,
                                                     "\\1\n#include <string.h>"
```

**Comment by asl** [comment]
Thanks. This is known issue. It will be fixed in 3.10.1

**Comment by asl** [comment]
Should be fixed in v3.10.1

**Comment by ilovezfs** [comment]
Thanks for the heads up: https://github.com/Homebrew/homebrew-science/pull/5207.
